addons/controls/README.md

```diff
@@ -0,0 +1,419 @@
+</center>
+
+<h1>Storybook Controls</h1>
+
+Storybook Controls is UI to interactively edit your component stories. Stories written for controls are:
+
+- **Convenient.** Auto-generate controls based on [React/Vue/Angular/etc.](#framework-support) components.
+- **Portable.** Reuse your interactive stories in documentation, tests, and even in designs.
+- **Rich.** Customize the controls and interactive data to suit your exact use case.
+
+Controls are built on top of [Storybook Args](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/formats/component-story-format/index.md#args-story-inputs), so you can reuse them in other contexts:
+
+- **Documentation.** 100% compatible with [Storybook Docs](https://github.com/storybookjs/storybook/tree/next/addons/docs).
+- **Testing.** Import stories directly into your [Jest](https://jestjs.io/) tests.
+- **Ecosystem.** Reuse stories in design/development tools that support [Component Story Format](https://medium.com/storybookjs/component-story-format-66f4c32366df).
+
+Controls replaces [Storybook Knobs](https://github.com/storybookjs/storybook/tree/master/addons/knobs). It incorporates lessons from years of supporting Knobs on tens of thousands of projects and dozens of different frameworks. We couldn't incrementally fix knobs, so we built a better version.
+
+<h2>Contents</h2>
+
+- [Get started](#get-started)
+- [Installation](#installation)
+- [Writing stories](#writing-stories)
+  - [No args](#no-args)
+  - [Auto-generated args](#auto-generated-args)
+  - [Custom controls args](#custom-controls-args)
+  - [Fully custom args](#fully-custom-args)
+  - [Template stories](#template-stories)
+- [Migrating from knobs](#migrating-from-knobs)
+  - [Knobs to auto-generated args](#knobs-to-auto-generated-args)
+  - [Knobs to manually-configured args](#knobs-to-manually-configured-args)
+- [Configuration](#configuration)
+  - [Control annotation](#control-annotation)
+  - [Parameters](#parameters)
+    - [expanded](#expanded)
+- [Framework support](#framework-support)
+- [FAQs](#faqs)
+  - [How will this replace addon-knobs?](#how-will-this-replace-addon-knobs)
+
+## Get started
+
+To get started with `addon-controls`:
+
+1. [Install the addon](#installation)
+2. [Write a story that uses Args](#writing-stories)
+3. [(optional) Configure the addon](#configuration)
+4. [(optional) Migrate knobs stories](#migrating-from-knobs)
+
+## Installation
+
+Controls requires [Storybook Docs](https://github.com/storybookjs/storybook/tree/next/addons/docs). If you're not using it already, please install that first.
+
+Next, install the package:
+
+```sh
+npm install @storybook/addon-controls -D # or yarn
+```
+
+And add it to your `.storybook/main.js` config:
+
+```js
+module.exports = {
+  addons: [
+    '@storybook/addon-docs'
+    '@storybook/addon-controls'
+  ],
+};
+```
+
+## Writing stories
+
+Once the addon is installed you should see a `Controls` tab in the addons panel.
```

Comment:
Tell the user what they'll accomplish in this section. 

"Let's see how to write stories that automatically generate controls... When you install the addon, you should see a Controls tab in the addons pannel."